lib/EmitVerilog/EmitVerilog.cpp

```diff
@@ -2034,10 +2063,31 @@ void ModuleEmitter::collectNamesEmitDecls(Block &block) {
     }
   }
 
-  if (!declsToEmit.empty())
+  if (!declsToEmit.empty() || rtlInstanceDeclaredWires)
     os << '\n';
 }
 
+bool ModuleEmitter::collectNamesEmitWires(rtl::RTLInstanceOp &op) {
+  for (size_t i = 0, e = op.getNumResults(); i < e; ++i) {
+    auto result = op.getResult(i);
+    StringRef wireName = addName(result, op.getResultName(i));
+
+    Type resultType = result.getType();
+    if (auto intType = resultType.dyn_cast<IntegerType>()) {
+      if (intType.getWidth() == 1) {
+        indent() << "wire " << wireName << ";\n";
+      } else {
+        indent() << "wire [" << intType.getWidth() - 1 << ":0] " << wireName
+                 << ";\n";
+      }
+    } else {
+      indent() << "// Type '" << resultType
```

Comment:
This should `emitOpError` as well so the client of the compiler knows the pass failed and the output had a problem.